### PR TITLE
`automat_dsc_configuration`: update sdk to `2022-08-08`

### PR DESCRIPTION
**Pull request by wuxu92**
based on #22404 

update to the new base-layered  sdk wil fix below acc tests (because of old sdk cannot process literal string reponse body):

```
TestAccAutomationDscConfiguration_basic
TestAccAutomationDscConfiguration_complete
TestAccAutomationDscConfiguration_requiresImport
TestAccAutomationDscNodeConfiguration_basic
TestAccAutomationDscNodeConfiguration_requiresImport
```

## Test Result:

```
--- PASS: TestAccAutomationDscNodeConfiguration_basic (130.93s)
--- PASS: TestAccAutomationDscNodeConfiguration_requiresImport (207.52s)
--- PASS: TestAccAutomationDscConfiguration_basic (185.03s)
--- PASS: TestAccAutomationDscConfiguration_requiresImport (197.80s)
--- PASS: TestAccAutomationDscConfiguration_complete (191.34s)
PASS
```

### CODE REVIEW

File: internal/services/automation/automation_dsc_configuration_resource_test.go [hunk 1]
```diff
@@ -61,7 +61,8 @@ func TestAccAutomationDscConfiguration_complete(t *testing.T) {
 				check.That(data.ResourceName).ExistsInAzure(r),
 			),
 		},
-		data.ImportStep(),
+		// not respond by Get API
+		data.ImportStep("log_progress", "hash", "source_version"),
```
Comment (tombuildsstuff) [inline internal/services/automation/automation_dsc_configuration_resource_test.go:65]: In this case, we'll need to add a Swagger issue about this and remove those fields from the Schema until these are returned.

File: internal/clients/client.go [hunk 2]
```diff
@@ -305,7 +305,9 @@ func (client *Client) Build(ctx context.Context, o *common.ClientOptions) error
 		return fmt.Errorf("building clients for Authorization: %+v", err)
 	}
 	client.Automanage = automanage.NewClient(o)
-	client.Automation = automation.NewClient(o)
+	if client.Automation, err = automation.NewClient(o); err != nil {
+		return fmt.Errorf("building clients for Autoamtion: %+v", err)
```
Comment (tombuildsstuff) [inline internal/clients/client.go:309]:
```suggestion
		return fmt.Errorf("building clients for Automation: %+v", err)
```

File: go.mod [hunk 3]
```diff
@@ -16,7 +16,7 @@ require (
 	github.com/google/go-cmp v0.5.9
 	github.com/google/uuid v1.3.0
 	github.com/hashicorp/go-azure-helpers v0.57.0
-	github.com/hashicorp/go-azure-sdk v0.20230623.1103505
+	github.com/hashicorp/go-azure-sdk v0.20230704.1182735
```
Comment (tombuildsstuff) [inline go.mod:19]: `go-azure-sdk` dependency updates should be done in their own PR so as not to conflict with other changes, would you mind splitting this one out?

File: website/docs/r/automation_dsc_configuration.html.markdown [hunk 4]
```diff
@@ -54,6 +54,14 @@ The following arguments are supported:
 
 * `tags` - (Optional) A mapping of tags to assign to the resource.
 
+* `log_progress` - (Optional) Whether enable the progress log option.
+
+* `hash_algorithm` - (Optional) Specifies the content hash algorithm used to hash the content.
+
+* `hash_value` - (Optional) Specifies the expected hash value of the content.
+
+* `source_version` - (Optional) Specifies the version of the content.
```
Comment (tombuildsstuff) [inline website/docs/r/automation_dsc_configuration.html.markdown:63]: (as above) we'll need to remove these for now

```suggestion
```

File: internal/services/automation/automation_dsc_configuration_resource.go [hunk 5]
```diff
@@ -173,6 +199,14 @@ func resourceAutomationDscConfigurationRead(d *pluginsdk.ResourceData, meta inte
...
 			d.Set("description", props.Description)
 			d.Set("state", string(pointer.From(props.State)))
+
+			if source := props.Source; source != nil {
+				d.Set("source_version", source.Version)
+				if hash := source.Hash; hash != nil {
+					d.Set("hash_algorithm", hash.Algorithm)
+					d.Set("hash_value", hash.Value)
+				}
+			}
```
Comment (tombuildsstuff) [inline internal/services/automation/automation_dsc_configuration_resource.go:209]: (as above) we'll need to remove these for now

```suggestion
```

File: internal/services/automation/automation_dsc_configuration_resource.go [hunk 6]
```diff
@@ -121,10 +141,16 @@ func resourceAutomationDscConfigurationCreateUpdate(d *pluginsdk.ResourceData, m
 	parameters := dscconfiguration.DscConfigurationCreateOrUpdateParameters{
 		Properties: dscconfiguration.DscConfigurationCreateOrUpdateProperties{
 			LogVerbose:  utils.Bool(logVerbose),
+			LogProgress: utils.Bool(d.Get("log_progress").(bool)),
```
Comment (tombuildsstuff) [inline internal/services/automation/automation_dsc_configuration_resource.go:144]: (as above) we'll need to remove this for now

```suggestion
```

File: internal/services/automation/automation_dsc_configuration_resource.go [hunk 7]
```diff
@@ -121,10 +141,16 @@ func resourceAutomationDscConfigurationCreateUpdate(d *pluginsdk.ResourceData, m
...
 			Description: utils.String(description),
 			Source: dscconfiguration.ContentSource{
 				Type:  pointer.To(dscconfiguration.ContentSourceTypeEmbeddedContent),
 				Value: utils.String(contentEmbedded),
+				Hash: &dscconfiguration.ContentHash{
+					Algorithm: d.Get("hash_algorithm").(string),
+					Value:     d.Get("hash_value").(string),
+				},
+				Version: utils.String(d.Get("source_version").(string)),
```
Comment (tombuildsstuff) [inline internal/services/automation/automation_dsc_configuration_resource.go:153]: (as above) we'll need to remove these for now

```suggestion
```

File: internal/services/automation/automation_dsc_configuration_resource.go [hunk 8]
```diff
@@ -75,6 +75,26 @@ func resourceAutomationDscConfiguration() *pluginsdk.Resource {
...
+
+			"hash_value": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+			},
+
+			"source_version": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+			},
```
Comment (tombuildsstuff) [inline internal/services/automation/automation_dsc_configuration_resource.go:96]: (as above) we'll need to remove these for now

```suggestion
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @wuxu92 

Thanks for this PR - taking a look through here it appears that we're going to want to split the changes in this PR though:

* `hashicorp/go-azure-sdk` tends to be updated regularly, as such can we split the SDK upgrade out of this PR?
* We'll need to remove the new fields which aren't returned from the API - and open an API tracking that, a field needs to be returned from the API for us to support it in Terraform, else this breaks `import` support - so we'll need that fixed before we can re-introduce those.

However if we can make those changes then this should otherwise be good to go 👍 

Thanks!

**Comment by wuxu92** [comment]
@tombuildsstuff thanks! the go-azure-sdk PR in #22404 . This PR updated to only update the dscconfiguration to 2022-08-08, new fields logic removed and the API issue tracked by https://github.com/Azure/azure-rest-api-specs/issues/24728

File: internal/services/automation/automation_dsc_configuration_resource_test.go [hunk 9]
```diff
@@ -61,7 +61,6 @@ func TestAccAutomationDscConfiguration_complete(t *testing.T) {
 				check.That(data.ResourceName).ExistsInAzure(r),
 			),
 		},
-		data.ImportStep(),
```
Comment (tombuildsstuff) [inline internal/services/automation/automation_dsc_configuration_resource_test.go:64]: we should be able to leave this one in?

```suggestion
		data.ImportStep(),
```

**Review by tombuildsstuff — APPROVED** [review]
One comment around adding back in the `Import` check for the `Complete` test (which I'm going to push a commit for) but this otherwise LGTM - thanks for making those changes @wuxu92

**Comment by tombuildsstuff** [comment]
(rebased this atop of `main` to remove the vendor changes made outside of this PR too)

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2023-07-07 at 15 22 03
